src/Classes/ProfilePage.php

```diff
@@ -0,0 +1,1020 @@
+				}
+			} catch ( CheevosException $e ) {
+				wfDebug( "Encountered Cheevos API error getting all achievements." );
+			}
+		}
+		$achievements = CheevosAchievement::correctCriteriaChildAchievements( $achievements );
+
+		$progresses = [];
+		if ( $type === 'general' ) {
+			try {
+				$progresses = Cheevos::getAchievementProgress(
+					[
+						'site_key' => $dsSiteKey,
+						'earned' => true,
+						'special' => false,
+						'shown_on_all_sites' => true,
+						'limit' => $limit,
+					],
+					$this->user
+				);
+			} catch ( CheevosException $e ) {
+				wfDebug( "Encountered Cheevos API error getting Achievement Progress." );
+			}
+		}
+
+		if ( $type === 'special' ) {
+			try {
+				$progresses =
+					Cheevos::getAchievementProgress(
+						[
+							'earned' => true,
+							'special' => true,
+							'shown_on_all_sites' => true,
+							'limit' => $limit,
+						],
+						$this->user
+					);
+			} catch ( CheevosException $e ) {
+				wfDebug( "Encountered Cheevos API error getting Achievement Progress." );
+			}
+		}
+		[ $achievements, $progresses ] =
+			CheevosAchievement::pruneAchievements( [ $achievements, $progresses ], true, true );
+
+		if ( empty( $progresses ) || !is_array( $progresses ) ) {
+			return [ '', 'isHTML' => true ];
+		}
+
+		$output = '<div class="section achievements">';
+		$output .= '<h4>' . $parser->recursiveTagParse( wfMessage( 'achievements-' . $type )->text() ) . '</h4>';
+		$count = 0;
+		foreach ( $progresses as $progress ) {
+			if ( !isset( $achievements[$progress->getAchievement_Id()] ) ) {
+				continue;
+			}
+			$ach = $achievements[$progress->getAchievement_Id()];
+			$output .= Html::rawElement(
+				'div',
+				[
+					'class'	=> [ 'icon', $type ]
+				],
+				Html::element(
+					'img',
+					[
+						'src'	=> $ach->getImageUrl(),
+						'title'	=> $ach->getName() . "\n" . $ach->getDescription()
+					]
+				) . ( $type === 'special' ? Html::rawElement(
+					'span',
+					[ 'title' => $ach->getName( $progress->getSite_Key() ) . "\n" . $ach->getDescription() ],
+					$ach->getName( $progress->getSite_Key() )
+				) : null )
+			);
+			/**
+			 * Workaround: achievements-service is not respecting limit param for progress. See:
+			 * http://a.poz-dev.k8s.wikia.net/achievements-service/v2/achievements/progress/?user_id=37714864&limit=2
+			 */
+			$count++;
+			if ( $limit && $count === $limit ) {
+				break;
+			}
+		}
+		$output .= '</div>';
+		return [ $output, 'isHTML' => true ];
+	}
+
+	/**
+	 * Performs the work for the parser tag that displays the user's level (based on wikipoints)
+	 *
+	 * @param mixed &$parser parser reference
+	 * @return mixed array with HTML string at index 0 or an HTML string
+	 */
+	public function userLevel( &$parser ) {
+		$userPoints = PointsDisplay::getWikiPointsForRange( $this->user );
+
+		$levelDefinitions = $this->config->get( 'PointsLevels' );
+
+		if ( !is_array( $levelDefinitions ) || empty( $levelDefinitions ) ) {
+			return '';
```

Comment:
Probably this was expected here, as '' was many times changed to construct below in other files:
```
return [
	'',
	'isHTML' => true,
];

Reply:
This code should be effectively dead as `PointsLevels` is defined as array in `extension.json` and I didn't find `PointsLevels` var in wikiConfig.
I will left this one as is to avoid additional retest

Reply:
Submodule has been bumped in https://github.com/Wikia/unified-platform/pull/12976/commits/d35bfe4d6ec66b2bb7ebcbb81133048437bde776